libs/clients/national-registry/v2/src/lib/nationalRegistryClient.service.ts

```diff
@@ -1,33 +1,130 @@
 import { Injectable } from '@nestjs/common'
 
-import { FetchError } from '@island.is/clients/middlewares'
+import { AuthMiddleware, User, Auth } from '@island.is/auth-nest-tools'
 
-import { EinstaklingarApi } from '../../gen/fetch'
+import { ApiResponse, EinstaklingarApi } from '../../gen/fetch'
 import { formatIndividualDto, IndividualDto } from './types/individual.dto'
+import {
+  CohabitationDto,
+  formatCohabitationDto,
+} from './types/cohabitation.dto'
+import {
+  formatResidenceHistoryEntryDto,
+  ResidenceHistoryEntryDto,
+} from './types/residence-history-entry.dto'
+import { FamilyDto, formatFamilyDto } from './types/family.dto'
+import { BirthplaceDto, formatBirthplaceDto } from './types/birthplace.dto'
+import { CitizenshipDto, formatCitizenshipDto } from './types/citizenship.dto'
 
 @Injectable()
 export class NationalRegistryClientService {
   constructor(private individualApi: EinstaklingarApi) {}
 
   async getIndividual(nationalId: string): Promise<IndividualDto | null> {
-    const response = await this.individualApi
-      .einstaklingarGetEinstaklingurRaw({
+    const individual = await this.handleModernMissingData(
+      this.individualApi.einstaklingarGetEinstaklingurRaw({
         id: nationalId,
-      })
-      .catch(this.handleError)
-
-    const individual =
-      response === null || response.raw.status === 204
-        ? null
-        : await response.value()
+      }),
+    )
 
     return formatIndividualDto(individual)
   }
 
-  private handleError(error: FetchError): null {
-    if (error.status === 404) {
+  async getCustodyChildren(parentUser: User): Promise<string[]> {
+    const response = await this.handleLegacyMissingData(
+      this.individualApi
+        .withMiddleware(new AuthMiddleware(parentUser))
+        .einstaklingarGetForsjaRaw({ id: parentUser.nationalId }),
+    )
+    return response || []
+  }
+
+  async getOtherCustodyParents(
+    parentUser: User,
+    childId: string,
+  ): Promise<string[]> {
+    const response = await this.handleLegacyMissingData(
+      this.individualApi
+        .withMiddleware(new AuthMiddleware(parentUser))
+        .einstaklingarGetForsjaForeldriRaw({
+          id: parentUser.nationalId,
+          barn: childId,
+        }),
+    )
+    return response || []
+  }
+
+  async getCohabitationInfo(
```

Comment:
I would use Maritial Status instead of cohabitation as the english for "hjúskaparstaða".
https://www.skra.is/english/forms/single-product/?productid=5d5773a3-137d-11e6-943e-005056851dd2
```suggestion
  async getMaritialStatusInfo(
```

Reply:
Hm ... this is the translation Þjóðskrá uses in their API, which I assume follows their formal [ontology](https://en.wikipedia.org/wiki/Ontology_(information_science)): 

https://island.is/throun/vefthjonustur/SVNfR09WXzY1MDM3NjA2NDlfU0tSQS1Qcm90ZWN0ZWRfRWluc3Rha2xpbmdhcg#operation/Einstaklingar_GetHjuskapur

There are examples of both naming conventions in our monorepo. I would lean toward using their naming if I hadn't already renamed some properties to fit OIDC naming. 😇

Reply:
🙈